[Intel MKL] Suppressing auto_mixed_precision_test for non-avx512f machine (broadwell)

noim210 opened the pull request:
[Intel MKL] Suppressing auto_mixed_precision_test for non-avx512f machine like broadwell.

noim210 committed:
[Intel MKL] Suppressing auto_mixed_precision_test for non-avx512f machine like broadwell

penpornk commented on tensorflow/python/framework/test_util.py:
```diff
@@ -1802,6 +1803,12 @@ def decorated(self, *args, **kwargs):
 
 
+# The description is just for documentation purposes.
+def disable_nonAVX512f(description):  # pylint: disable=unused-argument
```
We would still want to run CUDA tests if CPU has CUDA but doesn't have AVX-512f. Maybe it's better to add a condition to `_maybe_skip` instead:
https://github.com/tensorflow/tensorflow/blob/00476037681d6e10f8c0bca83251551134b169f8/tensorflow/python/grappler/auto_mixed_precision_test.py#L372-L373

noim210 replied:
Hi Penporn, Sorry for the delay, I was in vacation. Did you mean to use _maybe_skip from tensorflow/python/grappler/auto_mixed_precision_test.py. 
or you are suggesting to add "cuda" condition here like below from _maybe_skip : 
  def _maybe_skip(self, mode):
    if mode == 'cuda' and not test.is_gpu_available(cuda_only=True):
      self.skipTest('No GPU is available')
    if mode == 'mkl' and not test_util.IsMklEnabled():
      self.skipTest('MKL is not enabled')

noim210 replied:
Hi Penporn, Sorry for the delay, I was in vacation. Did you mean to use _maybe_skip from tensorflow/python/grappler/auto_mixed_precision_test.py.
or you are suggesting to add "cuda" condition here like below from _maybe_skip :
def _maybe_skip(self, mode):
if mode == 'cuda' and not test.is_gpu_available(cuda_only=True):
self.skipTest('No GPU is available')
if mode == 'mkl' and not test_util.IsMklEnabled():
self.skipTest('MKL is not enabled')

noim210 replied:
If you would like me to add lines inside "_maybe_skip" function, I am guessing if mode is 'cuda' and cuda is available , the test should run (no matter if avx-512f cpu or not?). Is it possible for you to clarify little more?

penpornk replied:
Hi Mazhar, Sorry for my late reply as well! I'm very far behind on github notifications lately.
I was thinking of the first option (use `_maybe_skip` and only modify `auto_mixed_precision_test.py`). But if you expect this to be used somewhere else as well, we can continue with `disable_nonAVX512f` (but not disable it for CUDA tests).

noim210 replied:
I have changed it . Would you please look at it to check if that is the thing you wanted me to do?

penpornk reviewed (changes requested):
Thank you for the PR and I'm so sorry for the delay!

gbaned commented:
@noim210  Can you please check @penpornk's comments and keep us posted ? Thanks!

gbaned commented:
@noim210  Any update on this PR? Please. Thanks!

noim210 committed:
Update auto_mixed_precision_test.py

noim210 committed:
Update auto_mixed_precision_test.py

noim210 committed:
Update auto_mixed_precision_test.py

gbaned commented:
@noim210  Can you please resolve conflicts? Thanks!

noim210 committed:
Merge branch 'master' into mazhar/auto_mixed_preci_bdw

penpornk commented:
(Changing the tag back to awaiting_response since I already replied in a comment.)

noim210 committed:
fixing the conflict for building

noim210 committed:
updating the code after getting reviews about cuda test run

penpornk reviewed (changes requested):
Thank you for the changes! Yes, those are what I meant. 
There are unrelated changes in `//tensorflow/python/BUILD` (probably because the file is out of date). Could you please sync to the latest commit? Thank you!

noim210 commented:
It looks like somebody refractor the code and move some parts from python/xx folder to python/util folder. That's why "`//tensorflow/python/BUILD`" file has some changes. 
The current changes what you are seeing, that are not my change, I just make sure current-master file and my branch's "`//tensorflow/python/BUILD`" file is same. That's why those changes are necessary (again to match with master).

penpornk commented:
The changes you have don't make the file up-to-date, though. They are undoing the newer changes to the file. For example, https://github.com/tensorflow/tensorflow/blob/2f6535b09c258f021b7bff50266348c0d1dbfb13/tensorflow/python/BUILD#L1594
This line was introduced yesterday and is still there in the most recent commit. Your commit is deleting it. That's why I think your file is not up-to-date.

noim210 commented:
That's interesting. My master is saying those files need to be deleted. Anyway, thank you for sharing the file link. I will make sure my file is same as your link.

noim210 committed:
update the file with new master

noim210 committed:
Merge branch 'master' into mazhar/auto_mixed_preci_bdw

noim210 commented:
I fixed the problem. Actually, my master might be older or I forget to merge my master. 
Now I pull master and updated it accordingly.
Thank you for helping me fix this.

penpornk commented on tensorflow/python/util/BUILD:
```diff
@@ -76,6 +76,7 @@ tf_python_pybind_extension(
         "//tensorflow/python:pybind11_lib",
         "//third_party/python_runtime:headers",
         "@pybind11",
+        "//tensorflow/core/platform:platform_port",
```
The list is sorted alphabetically. Could you please move this line to before `//tensorflow/python:pybind11_lib",`? Thank you very much!

noim210 replied:
thank you for pointing out. Let me do this now.

noim210 replied:
I fixed that. If you find any more required fix, please let me know :)

penpornk reviewed (changes requested):
Thank you for the quick fix! 
One last change please. Sorry I didn't comment on this earlier. I thought it was from merging.

noim210 committed:
fixing the order to make the lines lexicographic

penpornk reviewed (approved):
Thank you again for the PR and for your patience!
